[DM-4286] Update Fragment library with information about software type in device profiles (#3678)
SoftwareAG/c8y-docs: content/device-integration/fragment-library-bundle/device-profile.md

</tbody>
</table>

When a device receives a `c8y_Profile` operation it should announce the target profile in its own managed object first.

```http
PUT /inventory/managedObjects/<deviceId>
```

```json
{
  "c8y_Profile": {
    "profileName": "my profile",
    "profileId": "158751",
    "profileExecuted": false
  }
}
```

| Name            | Type    | Mandatory | Description                                                                            |
| --------------- | ------- | --------- | -------------------------------------------------------------------------------------- |
| profileName     | string  | Yes       | Name of the device profile                                                             |
| profileId       | string  | Yes       | The ID reference of the device profile object                                          |

> | profileExecuted | boolean | Yes       | Indicator showing if the profile has been applied fully. Must be false in this context |

Reviewer: ```suggestion
| profileExecuted | boolean | Yes       | Indicator showing if the profile has been applied fully; must be false in this context |
```

```http
PUT /inventory/managedObjects/<deviceId>
```

```json
{
  "c8y_Profile": {
    "profileName": "my profile",
    "profileId": "158751",
    "profileExecuted": true
  }
}
```

| Name            | Type    | Mandatory | Description                                                                           |
| --------------- | ------- | --------- | ------------------------------------------------------------------------------------- |
| profileName     | string  | Yes       | Name of the device profile                                                            |
| profileId       | string  | Yes       | The ID reference of the device profile object                                         |
| profileExecuted | boolean | Yes       | Indicator showing if the profile has been applied fully; must be true in this context |

The device is expected to perform the following actions:

1. Set operation status to EXECUTING
2. Set the `c8y_Profile` fragment in the device’s own managed object with profileExecuted = false

> 3. Install firmware if included and complete installation by updating the `c8y_Firmware` fragment in its own managed object

Reviewer: ```suggestion
3. Install firmware if included and complete the installation by updating the `c8y_Firmware` fragment in its own managed object
```

> 4. Install software if included and complete installation by updating the device installed software information, e.g. by [setting software list](/smartrest/mqtt-static-templates/#116) or by [setting advanced software list](/smartrest/mqtt-static-templates/#140)

Reviewer: ```suggestion
4. Install software if included and complete the installation by updating the device installed software information, for example, by [setting software list](/smartrest/mqtt-static-templates/#116) or by [setting advanced software list](/smartrest/mqtt-static-templates/#140)
```

> 5. Install configuration if included and complete installation by updating the `c8y_Configuration_<type>` fragment for each configuration in its own managed object

Reviewer: ```suggestion
5. Install configuration if included and complete the installation by updating the `c8y_Configuration_<type>` fragment for each configuration in its own managed object
```

6. Set the `c8y_Profile` fragment in the device's own managed object with profileExecuted = true
7. Set the operation status to SUCCESSFUL

**SmartREST example**

In addition to the static templates for firmware, software, and configuration provided by {{< product-c8y-iot >}} there are specific templates available for handling device profiles. The 527 static response template is designed to receive the operation. The 121 static template can be used to set the current state of device profile:

1. Receive `c8y_DeviceProfile` operation <br>

```
527,DeviceSerial,$FW,device_fw,1.0.1,https://cumulocity.com/fw,false,,$SW,curl,2.3.4,http://my.url.com,install,cumulocity_agent,1.2.3,https://cumulocity.com/agent,install,$CONF,http://cumulocity.com/conf,ssh_conf,https://demos.cumulocity.com/inventory/binaries/156719,agent_conf
```

> Template `527` is triggered whenever a `c8y_DeviceProfile` operation is created and will not include the software items types.

Reviewer: ```suggestion
Template `527` is triggered whenever a `c8y_DeviceProfile` operation is created and does not include the software item types.
```
